### PR TITLE
Add self import inspection

### PR DESCRIPTION
**Pull request by eugene-auduchinok**
I had a trouble with testing, so the first commit was for review only.
Implements #1350.

### CODE REVIEW

File: tests/com/goide/quickfix/GoSelfImportQuickFixTest.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  @Override
+  protected boolean isWriteActionRequired() {
+    return false;
+  }
+
+  public void testRemoveSelfImport() {
+    myFixture.configureByFile("b.go");
+    //applySingleQuickFix("Remove self import");
```
Comment (zolotov) [inline tests/com/goide/quickfix/GoSelfImportQuickFixTest.java]: ?

File: src/com/goide/inspections/GoSelfImportQuickFix.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import com.intellij.psi.PsiReference;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.containers.ContainerUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+public class GoSelfImportQuickFix extends LocalQuickFixBase {
+
+  protected GoSelfImportQuickFix(@NotNull String name) {
```
Comment (zolotov) [inline src/com/goide/inspections/GoSelfImportQuickFix.java]: `name` has always constant value

File: src/com/goide/inspections/GoSelfImportQuickFix.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    super(name);
+  }
+
+  @Override
+  public void applyFix(@NotNull final Project project, @NotNull ProblemDescriptor descriptor) {
+    final PsiElement element = descriptor.getPsiElement();
+    final PsiFile file = element != null ? element.getContainingFile() : null;
+    if (!(element instanceof GoImportSpec && file instanceof GoFile)) return;
+
+    final Map<GoReferenceExpressionBase, String> usages = ContainerUtil.newHashMap();
```
Comment (zolotov) [inline src/com/goide/inspections/GoSelfImportQuickFix.java]: Why don't just use GoDeleteQuickFix? Why do you need all this untested stuff with replacing?

File: src/com/goide/inspections/GoSelfImportQuickFix.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      public void run() {
+        for (Map.Entry<GoReferenceExpressionBase, String> usage : usages.entrySet()) {
+          GoReferenceExpressionBase expression = usage.getKey();
+          expression.replace(expression instanceof GoReferenceExpression
+                                      ? GoElementFactory.createReferenceExpression(project, usage.getValue())
+                                      : GoElementFactory.createTypeReferenceExpression(project, usage.getValue()));
+        }
+        GoImportDeclaration importDeclaration = PsiTreeUtil.getParentOfType(element, GoImportDeclaration.class);
+        assert importDeclaration != null;
+        PsiElement elementToDelete = importDeclaration.getImportSpecList().size() == 1 ? importDeclaration : element;
```
Comment (zolotov) [inline src/com/goide/inspections/GoSelfImportQuickFix.java]: This logic could be implemented in GoFile#deleteImport or something like that

**Comment by eugene-auduchinok** [comment]
Simplified.

File: src/com/goide/runconfig/testing/GoTestFinder.java [hunk 5]
```diff
@@ -46,6 +46,10 @@ public static boolean isTestFile(@Nullable VirtualFile file) {
     return file != null && file.getFileType() == GoFileType.INSTANCE && file.getNameWithoutExtension().endsWith(GoConstants.TEST_SUFFIX);
   }
 
+  public static boolean isTestPackageName(@Nullable String packageName) {
```
Comment (zolotov) [inline src/com/goide/runconfig/testing/GoTestFinder.java]: rebase master

File: src/com/goide/psi/impl/GoElementFactory.java [hunk 6]
```diff
@@ -111,8 +111,14 @@ public static GoStatement createShortVarDeclarationStatement(@NotNull Project pr
...
 
   @NotNull
-  public static GoReferenceExpression createVarReference(@NotNull Project project, @NotNull String name) {
+  public static GoReferenceExpression createReferenceExpression(@NotNull Project project, @NotNull String name) {
     GoFile file = createFileFromText(project, "package a; var a = " + name);
     return PsiTreeUtil.findChildOfType(file, GoReferenceExpression.class);
   }
+
+  @NotNull
+  public static GoTypeReferenceExpression createTypeReferenceExpression(@NotNull Project project, @NotNull String name) {
```
Comment (zolotov) [inline src/com/goide/psi/impl/GoElementFactory.java]: unrelated and unused, you can keep it but avoid this in future

File: src/com/goide/runconfig/testing/GoTestFinder.java (hunk 5 above)
Comment (eugene-auduchinok) [inline src/com/goide/runconfig/testing/GoTestFinder.java]: Done.

**Pull request merged**